### PR TITLE
[#2671] fix QR code appearance in dark mode

### LINKED ISSUE #2671: QR-code for enabling MFA unreadable in dark mode
### Product versie / Product version

2.0.3

### Omschrijf het probleem / Describe the bug

A collegue just tried to login to the admin portal for the first time, and noticed she couldn't scan the QR-code in dark mode:
image
(QR code blurred for obvious reasons)

This only goes for the QR-code used for the initial setup: Upon investigating using Windows' native dark mode (no plugins), at first I couldn't reproduce it: The QR-codes in the TOTP tab of the admin panel -do- show with a white background.

### Stappen om te reproduceren / Steps to reproduce

1. Set Windows (10) to dark mode
image

2. Delete current (if any) TOTP device from the OF admin-panel

3. Login and try to set the TOTP device anew

### Tasks

* [ ] Enforce white background for QR-code (so also in dark mode)
* [ ] Make sure there is some white padding (10px?) around the QR-code

### PR DESCRIPTION
**Pull request by pi-sigma**
Fixes open-formulieren/open-forms#2671

* added `style` attribute to the `img` tag with the QR code to add white background

### CODE REVIEW

**Comment by pi-sigma** [comment]
I changed the platform for GitHub actions to `ubuntu-20.04` because of test failures that resulted from a lack of support for certain Python versions (discussion of this issue [here](https://github.com/actions/setup-python/issues/543) and [here](https://github.com/actions/setup-python/issues/544). Not sure if this is the right approach.

File: two_factor/templates/two_factor/admin/setup.html [hunk 1]
```diff
@@ -20,7 +20,7 @@
         smartphone to scan the QR code below. For example, use Google
         Authenticator. Then, enter the token generated by the app.
         {% endblocktrans %}</p>
-    <p><img class="centered-image" src="{{ QR_URL }}" alt="QR Code" /></p>
+    <p><img class="centered-image" style="background-color: white" src="{{ QR_URL }}" alt="QR Code" /></p>
```
Comment (sergei-maertens) [inline two_factor/templates/two_factor/admin/setup.html]: this needs to be fixed in the CSS, not with inline styles as Content-Security Policies will block this (inline styles are typically blocked)

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ jobs:
       - name: Generate coverage XML report
         run: coverage xml
       - name: Codecov
-        uses: codecov/codecov-action@v1.0.10
+        uses: codecov/codecov-action@v1.5.2
```
Comment (sergei-maertens) [inline .github/workflows/ci.yml]: wasn't there a v3 already? I think we use that in the open forms backend

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ jobs:
     continue-on-error: ${{ matrix.experimental == 'true' }}
     strategy:
       matrix:
-        python-version: ['3.6', '3.7', '3.8']
+        python-version: ['3.8', '3.9', '3.10', '3.11']
         django-version: ['2.2', '3.0', '3.1']
```
Comment (sergei-maertens) [inline .github/workflows/ci.yml]: 2.2 is end of life, so are 3.0 and 3.1. We should only support 3.2 (and 4.2 in a couple of weeks when it releases)

File: tox.ini [hunk 4]
```diff
@@ -2,10 +2,10 @@
...
 minversion = 1.8
 envlist =
-    py{35,36,37,38}-dj22-{normal,yubikey,custom_user},
-    py{36,37,38}-dj30-{normal,yubikey,custom_user},
-    py{36,37,38}-dj31-{normal,yubikey,custom_user},
-    py{36,37,38}-djmaster-{normal,yubikey,custom_user}
+    py{38,39,310,311}-dj22-{normal,yubikey,custom_user},
+    py{38,39,310,311}-dj30-{normal,yubikey,custom_user},
+    py{38,39,310,311}-dj31-{normal,yubikey,custom_user},
+    py{38,39,310,311}-djmaster-{normal,yubikey,custom_user}
```
Comment (sergei-maertens) [inline tox.ini]: this build matrix also needs updating for proper django 3.2+ env, take a look at https://github.com/maykinmedia/drf-polymorphic/blob/main/tox.ini#L3 for example

File: tox.ini [hunk 5]
```diff
@@ -24,10 +24,10 @@ VARIANT =
 
 [gh-actions]
 python =
-    3.5: py35
-    3.6: py36
-    3.7: py37
     3.8: py38, mypy
+    3.9: py39, mypy
+    3.10: py310, mypy
+    3.11: py311, mypy
```
Comment (sergei-maertens) [inline tox.ini:30]: doesn't need mypy afaik, I don't know what the mypy version targets

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]

**Comment by sergei-maertens** [comment]
decided against updating this fork and instead we'll tackle things in OF directly.

**Pull request closed without merging**